### ISSUE TITLE
New CTRL + scrollwheel to zoom works really well, some ideas for UX improvement of the feature . . .

### ISSUE THREAD

**Issue by charlesmlamb** [godotengine/godot-proposals#3340]
### Describe the project you are working on

A barbarian 2.5D, with lore . . .

### Describe the problem or limitation you are having in your project

First of all, even having zoom for graph edit on scroll-wheel, it's amazing, major improvement . . . However, in Godot 4 it happens with no CTRL, in Godot 3.4 it happens with scroll-wheel, and in 3.4, no button + scroll-wheel = vertical jump, SHIFT + scroll-wheel = horizontal jump . . .

I had some problems with scroll-wheel auto-triggering a zoom, or jump, bec. of a sensitive mouse, that also seems to have been solved . . . My suggestion, remove the vertical, horizontal jumps, make perhaps default scroll-wheel no button zoom, and add feature, to set it to CTRL + scroll-wheel, for users with a sensitive mouse, or scroll-wheel . . . Right now, graph edits are getting much better, difficult to express how this will benefit Godot users and, as someone who has had much benefit from visual script, can now write code, after understanding the engine, over-all the improvement is just 10 / 10 thanks . . .

Wanted to ask for these last details, at least for me the horizontal, vertical jumps are just confusing, end up in places I wouldn't think, one way might to make it a slow-animated jump, sort of it moves, linearly, over 1 second, to new place or, faster, or perhaps remove these, or consider using scroll-wheel, shift-scroll wheel for other stuff, for instance to scroll grid size, or so . . .

It makes little sense to use scroll-wheel for horizontal jumps, as wheel is most often vertical, in my opinion it makes little sense . . . Vertical jump makes more sense, for some reason it isn't ideal . . . I think it's amazing we got these new features, it's amazing . . . I'm hoping it might be possible, to use them a bit better . . . .

For instance, I'd think using SHIFT + scroll-wheel might be better in the 3D editor, to have a slow zoom, and fast zoom, so scroll-wheel is normal zoom speed, SHIFT + scroll-wheel is slow zoom, CTRL + scroll-wheel is fast zoom . . .

One could also make SHIFT + scroll-wheel into a fast pan, with 2x pan, of graph edit canvas, really think there has been amazing features added, it's bec. a superb feature, and well this will benefit new coders, perhaps artists, prototype work, visual shaders, even if rumor is true, we get a particle graph edit, it's time to make this a neat, well - rounded feature, not really sure what to use the buttons, or scroll-wheel for, imo. it's brimming with possibilities, hope you'll review this and, if possible, add the finishing touches, so it makes bit more sense, and perhaps adds a bit more, to the editor, or UX . . .

9.5 / 10 already, so no complaints, it's more getting it to 9.8 / 10, big thanks . . .

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Well, right now it feels the buttons are perhaps wasted, having a 2 x, or 3 x pan speed, or something, I think makes more sense, well, can't complain, everything is getting better, so thanks . . . .

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Well, like above, something that makes better or, perhaps full use of the scroll-wheel, or so . . .

Well, no real complaints . . . .

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Can't re-code engine . . .

### Is there a reason why this should be core and not an add-on in the asset library?

Well, atm. there are just incredible improvements, in UX, I feel these last elements, might be better used though, atm. it's well amazing . . . .

**Comment by SIsilicon** [comment]
Unrelated to the proposal, I notice that you respond to the second last question (If this enhancement will not be used often, can it be worked around with a few lines of script?) stating that you can't implent in script yourself. But what the question really meant was "If _anyone, or someone with good programming experience_ could implement this in a few lines of code?"

**Comment by charlesmlamb** [comment]
No, it's I'm so beginner, don't understand what a good coder can do :DD <3 . .
